servicetalk-data-jackson/src/main/java/io/servicetalk/data/jackson/JacksonStreamingSerializer.java

```diff
@@ -0,0 +1,412 @@
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.BooleanNode;
+import com.fasterxml.jackson.databind.node.NullNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+
+import java.io.IOException;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.List;
+import javax.annotation.Nullable;
+
+import static com.fasterxml.jackson.databind.node.JsonNodeFactory.instance;
+import static io.servicetalk.concurrent.internal.EmptySubscriptions.EMPTY_SUBSCRIPTION;
+import static io.servicetalk.data.jackson.JacksonSerializer.doSerialize;
+import static java.util.Collections.singletonList;
+import static java.util.function.Function.identity;
+
+/**
+ * Serialize a stream of JSON objects.
+ * @param <T> The type of objects to serialize.
+ */
+final class JacksonStreamingSerializer<T> implements StreamingSerializerDeserializer<T> {
+    private final ObjectWriter writer;
+    private final ObjectReader reader;
+
+    JacksonStreamingSerializer(ObjectMapper mapper, Class<T> clazz) {
+        writer = mapper.writerFor(clazz);
+        reader = mapper.readerFor(clazz);
+    }
+
+    JacksonStreamingSerializer(ObjectMapper mapper, TypeReference<T> typeRef) {
+        writer = mapper.writerFor(typeRef);
+        reader = mapper.readerFor(typeRef);
+    }
+
+    JacksonStreamingSerializer(ObjectMapper mapper, JavaType type) {
+        writer = mapper.writerFor(type);
+        reader = mapper.readerFor(type);
+    }
+
+    @Override
+    public Publisher<Buffer> serialize(final Publisher<T> toSerialize, final BufferAllocator allocator) {
+        return toSerialize.map(t -> {
+            Buffer buffer = allocator.newBuffer();
+            doSerialize(writer, t, buffer);
+            return buffer;
+        });
+    }
+
+    @Override
+    public Publisher<T> deserialize(final Publisher<Buffer> serializedData, final BufferAllocator allocator) {
+        return serializedData.liftSync(new DeserializeOperator<T>(reader)).flatMapConcatIterable(identity());
+    }
+
+    private static final class DeserializeOperator<T> implements PublisherOperator<Buffer, Iterable<T>> {
+        private final ObjectReader reader;
+
+        private DeserializeOperator(ObjectReader reader) {
+            this.reader = reader;
+        }
+
+        @Override
+        public Subscriber<? super Buffer> apply(final Subscriber<? super Iterable<T>> subscriber) {
+            final JsonParser parser;
+            try {
+                // TODO(scott): ByteBufferFeeder is currently not supported by jackson, and the current API throws
```

Comment:
Do we want this in now? Shall we open a radar?

Reply:
https://github.com/apple/servicetalk/issues/1711 ... this is long standing from 3+ years ago just cpy/pasta to use the new APIs.